### PR TITLE
♻️ Move db implementation from webserver/payments to packages/postgres-database

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Moves db implementations from webserver/payment plugin to the `simcore_postgres_database` library so it can be reused at the `payments` service `db` module. We have moved ONLY the functionality for one-time-payment

image

NOTE the following class used to report failures instead of raising exceptions. 
```python
@dataclass
class PaymentFailure:
    payment_id: str

    def __bool__(self):
        return False
```
I tried this in the spirit of https://github.com/pviafore/RobustPython/blob/master/code_examples/chapter4/sum_type.py

## Related issue/s

- prerequisite for https://github.com/ITISFoundation/osparc-simcore/pull/4743
- related to https://github.com/ITISFoundation/osparc-simcore/issues/4657

## How to test

- Driving test `packages/postgres-database/tests/test_models_payments_transactions.py`

## DevOps
None

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]

**Review by bisgaard-itis — APPROVED** [review]
Looks good to me.

File: packages/postgres-database/src/simcore_postgres_database/utils_payments.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+from . import errors
+from .models.payments_transactions import PaymentTransactionState, payments_transactions
+
+PaymentID: TypeAlias = str
+
+PaymentTransactionRow: TypeAlias = RowProxy
+
+
+@dataclass
+class PaymentFailure:
```
Comment (GitHK) [inline packages/postgres-database/src/simcore_postgres_database/utils_payments.py:19]: I would have expected this to be a pedantic model.

File: packages/postgres-database/tests/test_models_payments_transactions.py [hunk 2]
```diff
@@ -71,18 +79,24 @@ async def _init(payment_id: str):
         values["initiated_at"] = utcnow()
 
         # insert
-        await connection.execute(payments_transactions.insert().values(values))
+        await insert_init_payment_transaction(connection, **values)
```
Comment (GitHK) [inline packages/postgres-database/tests/test_models_payments_transactions.py:82]: please assert the return type, that it is not `PaymentAlreadyExists`

Comment (GitHK) [reply packages/postgres-database/tests/test_models_payments_transactions.py:82]: I would also add a test for when this fails

**Review by GitHK — APPROVED** [review]

File: packages/postgres-database/src/simcore_postgres_database/utils_payments.py (hunk 1 above)
Comment (pcrespov) [reply packages/postgres-database/src/simcore_postgres_database/utils_payments.py:19]: `pydantic` you mean :-)
smcor_postgres_database ha no pydantic library installed

File: packages/postgres-database/tests/test_models_payments_transactions.py (hunk 2 above)
Comment (pcrespov) [reply packages/postgres-database/tests/test_models_payments_transactions.py:82]: there is a test about it below: test_complete_transaction

**Pull request merged**